.github/workflows/main.yml

```diff
@@ -140,6 +140,27 @@ jobs:
             exit 7
           fi
 
+      - name: Check Image size
+        env:
+          IMAGE_ARCH: ${{ runner.arch == 'X64' && 'amd64' || 'arm64' }}
+          TOMCAT: tomcat${{ matrix.tomcat_major }}
+          JAVA: jre${{ matrix.java_major }}
+          DISTRIBUTION: ${{ matrix.base_image.flavor }}${{ matrix.base_image.major }}
+        run: |
+          IMAGE_PREV_SHA=$(docker manifest inspect ${IMAGE_REGISTRY_NAMESPACE}/${IMAGE_REPOSITORY}:${TOMCAT}-${JAVA}-${DISTRIBUTION} | jq -r '.manifests|map(select(.platform.architecture=="amd64"))[0].digest')
+          if [ -z "$IMAGE_PREV_SHA" ]; then
+            echo "No previous image found, skipping check"
+            exit 0
+          else echo "Previous image found: $IMAGE_PREV_SHA"
+            IMAGE_PREV_SIZE=$(docker manifest inspect ${IMAGE_REGISTRY_NAMESPACE}/${IMAGE_REPOSITORY}@${IMAGE_PREV_SHA} | jq '.layers|map(.size)|add')
+            IMAGE_NEW_SIZE=$(docker save local/${{ env.IMAGE_REPOSITORY }}:ci | gzip -c | wc -c)
```

Comment:
you can't use the same docker inspect approach for the locally built? I am a bit scared that in this way we are not really comparing the same thing

Reply:
I don't think I can do that as the other method is using manifest inspection and I don't have a manifest for an image that's local only.
Arguably this is just an estimation and we're not comparing like for like but I think that's all we need here